packages/next-swc/crates/core/tests/errors/server-actions/server-graph/1/output.js

```diff
@@ -1,9 +1,11 @@
-/* __next_internal_action_entry_do_not_use__ foo */ export function foo() {}
-import ensureServerEntryExports from "private-next-rsc-action-proxy";
+/* __next_internal_action_entry_do_not_use__ foo */ import __create_action_proxy__ from "private-next-rsc-action-proxy";
+export function foo() {}
+import ensureServerEntryExports from "private-next-rsc-action-validate";
 ensureServerEntryExports([
     foo
 ]);
 foo.$$typeof = Symbol.for("react.server.reference");
 foo.$$id = "ab21efdafbe611287bc25c0462b1e0510d13e48b";
 foo.$$bound = null;
 foo.$$with_bound = false;
```

Comment:
What is `$$with_bound`?

Reply:
Mostly implementation detail from early days that I should get rid of soon. It's for manipulating the shape of arguments based on if it's a transpiled action (inlined) or not (file-level).

In a "use server" file we can't transpile all the action functions as we don't know which function will be exported (it's a runtime thing). What we are doing is to wrap all exported fields. Because of that, we don't manipulate the action's argument list during compilation so if a function takes arguments, say `async function action(a, b, c)`, we need to pass these arguments in that shape when invoking the action.

But for inlined actions, since we have an array of bound values + arguments received, a short path (the initial commit of the transpiler) we were taking is to compile the action to `async function action(closure, closure_value_1 = closure[0], a = closure[1], b = closure[2], c = closure[3])` and then calling the action with `action($$bound)`. 

This needs to be fixed as `async function action(closure_value_1, a, b, c)` to make it unified.

Reply:
Refactored and it's now gone.